# 1. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[0] + s[2] + s[4] + s[6])
```

Ответ: `aceg`

# 2. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[0]*3 + s[-1]*3 + s[3]*2 + s[3]*2)
```

Ответ: `aaagggdddd`


# 3. Что покажет приведенный ниже фрагмент кода?
```
s = '01234567891011121314151617'
for i in range(0, len(s), 5):
    print(s[i], end='')
```

Ответ: `051217`


# 4. Дополните приведенный код, используя индексатор, так чтобы он вывел символ запятой.

In [1]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[7])

,


# 5. Дополните приведенный код, используя индексатор, так чтобы он вывел символ w.

In [2]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[-10])

w


# 6. В столбик 1
На вход программе подается одна строка. Напишите программу, которая выводит элементы строки с индексами 0, 2, 4, ... в столбик.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести элементы строки с индексами 0, 2, 4, ..., каждое на отдельной строке.

In [3]:
n = input()
m = len(n)
for i in range(0, m, 2):
    print(n[i])

abcdefghijklmnop
a
c
e
g
i
k
m
o


# 7. В столбик 2
На вход программе подается одна строка. Напишите программу, которая выводит в столбик элементы строки в обратном порядке.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести в столбик элементы строки в обратном порядке. 

In [4]:
n = input()
m = -len(n)
for i in range(-1, m - 1, -1):
    print(n[i])

abc
c
b
a


# 8. ФИО
На вход программе подаются три строки: имя, фамилия и отчество. Напишите программу, которая выводит инициалы человека.

`Формат входных данных`

На вход программе подаются три строки, каждая на отдельной строке.

`Формат выходных данных`

Программа должна вывести ФИО человека.

Примечание. Гарантируется, что имя, фамилия и отчество начинаются с заглавной буквы.

In [7]:
a, b, c = input(), input(), input()
print(a[0], b[0], c[0], sep='')

Гуев
Тимур
Ахсарбекович
ГТА


# 9. Цифра 1
На вход программе подается одна строка состоящая из цифр. Напишите программу, которая считает сумму цифр данной строки.

`Формат входных данных`

На вход программе подается одна строка состоящая из цифр.

`Формат выходных данных`

Программа должна вывести сумму цифр данной строки.

In [8]:
n = int(input())
a = str(n)
print(int(a[0]) + int(a[1]) + int(a[2]) + int(a[3]))

2514
12


# 10. Цифра 2
На вход программе подается одна строка. Напишите программу, которая выводит сообщение «Цифра» (без кавычек), если строка содержит цифру. В противном случае вывести сообщение «Цифр нет» (без кавычек).

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [9]:
n = input()
c = 0
for i in range(0, 10):
    if str(i) in n:
        c += 1
if c > 0:
    print("Цифра")
else:
    print("Цифр нет")

Hi 17 Python
Цифра


# 11. Сколько раз?
На вход программе подается одна строка. Напишите программу, которая определяет сколько раз в строке встречаются символы + и *.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести сколько раз встречаются символы  + и * в строке.

In [10]:
a = input()
c = 0
d = 0
for i in a:
    if i == "+":
        c += 1
for i in a:
    if i == "*":
        d += 1
print("Символ + встречается", c, "раз")
print("Символ * встречается", d, "раз")

bcd+a++++**31415
Символ + встречается 5 раз
Символ * встречается 2 раз


# 12. Одинаковые соседи
На вход программе подается одна строка. Напишите программу, которая определяет сколько в ней одинаковых соседних символов.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести количество одинаковых соседних символов.

In [11]:
a = input()
c = 0
for i in range(len(a) - 1):
    if a[i] == a[i + 1]:
        c +=1
print(c) 

aabbcc
3


# 13. Гласные и согласные
На вход программе подается одна строка с буквами русского языка. Напишите программу, которая определяет количество гласных и согласных букв.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести количество гласных и согласных букв.

Примечание. В русском языке 10 гласных букв (а, у, о, ы, и, э, я, ю, ё, е) и 21 согласная буква (б, в, г, д, ж, з, й, к, л, м, н, п, р, с, т, ф, х, ц, ч, ш, щ).

In [12]:
a = input()
n = ['а', 'у', 'о', 'ы', 'и', 'э', 'я', 'ю', 'ё', 'е', 'A', 'У', 'О', 'Ы', 'И', 'Э', 'Я', 'Ю', 'Ё', 'Е']
k = ['б', 'в', 'г', 'д', 'ж', 'з', 'й', 'к', 'л', 'м', 'н', 'п', 'р', 'с', 'т', 'ф', 'х', 'ц', 'ч', 'ш', 'щ', 'Б', 'В', 'Г', 'Д', 'Ж', 'З', 'Й', 'К', 'Л', 'М', 'Н', 'П', 'Р', 'С', 'Т', 'Ф', 'Х', 'Ц', 'Ч', 'Ш', 'Щ',]
gl = 0
sog = 0 
for i in range(len(a)):
    if a[i] in n:
        gl += 1
for i in range(len(a)):
    if a[i] in k:
        sog += 1
print("Количество гласных букв равно", gl)
print("Количество согласных букв равно", sog)

Вдохновение – это умение приводить себя в рабочее состояние!
Количество гласных букв равно 25
Количество согласных букв равно 24


# 14. Decimal to Binary
На вход программе подается натуральное число, записанное в десятичной системе счисления. Напишите программу, которая переводит данное число в двоичную систему счисления.

`Формат входных данных`

На вход программе подается одно натуральное число.

`Формат выходных данных`

Программа должна вывести число записанное в двоичной системе счисления.

In [13]:
n = int(input())
a = str()
while n > 0:
    a = str(n % 2) + a
    n //= 2
print(a)

5
101


# 15. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[2:5])
```

Ответ: `cde`

# 16. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[3:])
```

Ответ: `defg`

# 17. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[:3])
```

Ответ: `abc`


# 18. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[:])
```

Ответ: `abcdefg`


# 19. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[::-3])
```

Ответ: `gda`


# 20. Дополните приведенный код, используя срезы, так чтобы он вывел первые 12 символов строки s.

In [14]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[:12])

In 2010, som


# 21. Дополните приведенный код, используя срезы, так чтобы он вывел последние 9 символов строки s.

In [15]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[-9:])

o pizzas.


# 22. Дополните приведенный код, используя срезы, так чтобы он вывел каждый 7 символ строки s начиная от начала строки.

In [16]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[::7])

I,n top


# 23. Дополните приведенный код, используя срезы, так чтобы он вывел строку s в обратном порядке.

In [17]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[::-1])

.sazzip owt rof nioctiB k01 diap enoemos ,0102 nI


# 24. Палиндром
На вход программе подается одно слово, записанное в нижнем регистре. Напишите программу, которая определяет является ли оно палиндромом.

``Формат входных данных``

На вход программе подается одно слово в нижнем регистре.

``Формат выходных данных``

Программа должна вывести «YES», если слово является палиндромом и «NO» в противном случае.

Примечание. Палиндром читается одинаково в обоих направлениях, например слово «потоп».

In [18]:
n = input()
if n == n[::-1]:
    print("YES")
else:
    print("NO")

потоп
YES


# 25. Делаем срезы 1
На вход программе подается одна строка. Напишите программу, которая выводит:

* общее количество символов в строке;
* исходную строку повторенную 3 раза;
* первый символ строки;
* первые три символа строки;
* последние три символа строки;
* строку в обратном порядке;
* строку с удаленным первым и последним символом.

`Формат входных данных`

На вход программе подается одна строка, длина которой больше 3 символов.

`Формат выходных данных`

Программа должна вывести данные в соответствии с условием. Каждое значение выводится на отдельной строке.

In [19]:
n = input()
print(len(n), n * 3, n[0], n[:3], n[-3:], n[::-1], n[1:-1], sep="\n")

abcdefghijklmnopqrstuvwxyz
26
abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz
a
abc
xyz
zyxwvutsrqponmlkjihgfedcba
bcdefghijklmnopqrstuvwxy


# 26. Делаем срезы 2
На вход программе подается одна строка. Напишите программу, которая выводит:

* третий символ этой строки;
* предпоследний символ этой строки;
* первые пять символов этой строки;
* всю строку, кроме последних двух символов;
* все символы с четными индексами;
* все символы с нечетными индексами;
* все символы в обратном порядке;
* все символы строки через один в обратном порядке, начиная с последнего.

`Формат входных данных`
На вход программе подается одна строка, длина которой больше 5 символов.

`Формат выходных данных`

Программа должна вывести данные в соответствии с условием. Каждое значение выводится на отдельной строке.

In [20]:
n = input()
print(n[2], n[-2], n[:5], n[:-2], n[0::2], n[1::2], n[::-1], n[::-2], sep="\n")

abcdefghijklmnopqrstuvwxyz
c
y
abcde
abcdefghijklmnopqrstuvwx
acegikmoqsuwy
bdfhjlnprtvxz
zyxwvutsrqponmlkjihgfedcba
zxvtrpnljhfdb


# 27. Две половинки
На вход программе подается строка текста. Напишите программу, которая разрежет ее на две равные части, переставит их местами и выведет на экран.

`Формат входных данных`

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

Примечание. Если длина строки нечетная, то длина первой части должна быть на один символ больше.

In [21]:
n = input()
a = n[0:len(n) - len(n) // 2]
b = n[len(n) - len(n) // 2:len(n)]
print(b, a, sep="")

abcdefg
efgabcd


# 28. Что покажет приведенный ниже фрагмент кода?
```
s = 'i Learn Python language'
print(s.capitalize())
```

Ответ: `I learn python language`

# 29. Что покажет приведенный ниже фрагмент кода?
```
s = 'i LEARN Python LAnguaGE'
print(s.lower())
```

Ответ: `i learn python language`

# 30. Что покажет приведенный ниже фрагмент кода?
```
s = '$12344%^$#@!'
print(s.lower())
```

Ответ: `$12344%^$#@!`

# 31. Что покажет приведенный ниже фрагмент кода?
```
s1 = 'a'
s2 = s1.upper()
print(s1, s2)
```

Ответ: `a A`

# 32. Что покажет приведенный ниже фрагмент кода?
```
s = 'i LEARN Python LAnguaGE'
print(s.upper())
```

Ответ: `I LEARN PYTHON LANGUAGE`

# 33. Что покажет приведенный ниже фрагмент кода?
```
s = 'i LEARN Python LAnguaGE'
print(s.swapcase())
```

Ответ: `I learn pYTHON laNGUAge`

# 34. Заглавные буквы
На вход программе подается строка состоящая из имени и фамилии человека, разделенных одним пробелом. Напишите программу, которая проверяет, что имя и фамилия начинаются с заглавной буквы.

`Формат входных данных` 

На вход программе подается строка.

`Формат выходных данных`

Программа должна вывести «YES» если имя и фамилия начинаются с заглавной буквы и «NO» в противном случае.

Примечание. Строка содержит только буквы. 

In [22]:
n = input()
if n == n.title():
    print("YES")
else:
    print("NO")

chris alan
NO


# 35. sWAP cASE
На вход программе подается строка. Напишите программу, которая меняет регистр символов, другими словами замените все строчные символы заглавными и наоборот.

`Формат входных данных`

На вход программе подается строка.

`Формат выходных данных`

Программа должна вывести строку в соответствии с условием задачи.

In [23]:
n = input()
print(n.swapcase())

Swap Case
sWAP cASE


# 36. Хороший оттенок
На вход программе подается строка текста. Напишите программу, которая определяет является ли оттенок текста хорошим или нет. Текст имеет хороший оттенок, если содержит подстроку «хорош» во всевозможных регистрах.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести «YES» если текст имеет хороший оттенок и «NO» в противном случае.

Примечание. Текст содержащий хорош, ХОРОШ, Хорош, хОРОШ и т.д. имеет хороший оттенок.

In [24]:
n = input().lower()
if "хорош" in n:
    print("YES")
else:
    print("NO")

оыралоывало ХОРОШвмсва выарлво83кг834
YES


# 37. Нижний регистр
На вход программе подается строка. Напишите программу, которая подсчитывает количество буквенных символов в нижнем регистре.

`Формат входных данных `

На вход программе подается строка.

`Формат выходных данных`

Программа должна вывести количество буквенных символов в нижнем регистре.

In [25]:
n = input()
s = 0
a = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
for i in range(len(n)):
    if n[i] in a:
        s += 1
print(s)

gggggggg1212321ABDCEFCE
8


# 38. Что покажет приведенный ниже фрагмент кода?
```
s = 'aabbAAccDDaa'
s = s.lower()
print(s.count('a'))
```

Ответ: `6`

# 39. Что покажет приведенный ниже фрагмент кода?
```
s = 'www.stepik.org'
print(s.startswith('www'))
```

Ответ: `True`

# 40. Что покажет приведенный ниже фрагмент кода?
```
s = 'www.stepik.org'
print(s.endswith('.ru'))
```

Ответ: `False`


# 41. Что покажет приведенный ниже фрагмент кода?
```
s = 'I learn Python language. Python - awesome!'
print(s.find('Python'))
```

Ответ: `8`


# 42. Что покажет приведенный ниже фрагмент кода?
```
s = '     I learn Python language               '
print(s.strip())
```

Ответ: `I learn Python language`

# 43. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdababa'
print(s.replace('ab', '123'))
```

Ответ: `123cd123123a`

# 44. Количество слов
На вход программе подается строка текста, состоящая из слов, разделенных ровно одним пробелом. Напишите программу, которая подсчитывает количество слов в ней.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести количество слов.

Примечание 1. Строка текста не содержит пробелов в начале и конце.

Примечание 2. Используйте для решения задачи метод count.

In [26]:
n = input()
print(n.count(" ") + 1)

In 2010, someone paid 10k Bitcoin for two pizzas.
9


# 45. Минутка генетики
На вход программе подается строка генетического кода, состоящая из букв А (аденин), Г (гуанин), Ц (цитозин), Т (тимин). Напишите программу, которая подсчитывает сколько аденина, гуанина, цитозина и тимина входит в данную строку генетического кода.

`Формат входных данных `

На вход программе подается строка генетического кода, состоящая из символов А, Г, Ц, Т, а, г, ц, т.

`Формат выходных данных`

Программа должна вывести сколько гуанина, тимина, цитозина, аденина входит в данную строку генетического кода.

Примечание. Строка не содержит символов, кроме как А, Г, Ц, Т, а, г, ц, т.

In [27]:
n = input().lower()
print("Аденин:", n.count("а"))
print("Гуанин:" , n.count("г"))
print("Цитозин:", n.count("ц"))
print("Тимин:", n.count("т"))

АааГГЦЦцТТттт
Аденин: 3
Гуанин: 2
Цитозин: 3
Тимин: 5


# 46. Очень странные дела
Джим Хоппер с помощью радиоприемника пытается получить сообщение Оди. На приемник ему поступает nn различных последовательностей кода Морзе. Декодировав их, он получает последовательности из цифр и строчного латинского алфавита, при этом во всех сообщениях Оди содержится число 11, причем минимум 3 раза. Помогите определить Джиму количество сообщений от Оди.

`Формат входных данных`

В первой строке подаётся число nn – количество сообщений, в последующих nn строках вводятся строки, содержащие латинские строчные буквы и цифры.

`Формат выходных данных`

Программа должна вывести количество строк в которых содержится число 11 минимум 3 раза.

Примечание: Числа 11 необязательно должны быть разделены другими символами, нужно подсчитать вхождение последовательности символов "11", т.е. например в строке "111" содержится одна такая последовательность, в то время как в "1111" их уже две.

In [28]:
n = int(input())
d = 0
for i in range(n):
    a = input()
    if a.count("11") >= 3:
        d += 1
print(d)   

3
11helpme11jim11
avengers141414atta11ck
k1lg0re11111l
1


# 47. Количество цифр
На вход программе подается строка текста. Напишите программу, которая подсчитывает количество цифр в данной строке.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести количество цифр в данной строке.

In [29]:
s = input()
count = 0
for i in range(len(s)):
    if s[i].isnumeric():
        count += 1
print(count)

l33t 3301
6


# 48. .com or .ru
На вход программе подается строка текста. Напишите программу, которая проверяет, что строка заканчивается подстрокой .com или .ru.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести «YES» если введенная строка заканчивается подстрокой .com или .ru и «NO» в противном случае.

In [30]:
n = input()
a = n.endswith('.ru')
b = n.endswith('.com')
if a == False and b == False:
    print('NO')
else:
    print('YES')

www.google.com
YES


# 49. Самый частотный символ
На вход программе подается строка текста. Напишите программу, которая выводит на экран символ, который появляется наиболее часто.

`Формат входных данных `

На вход программе подается строка текста. Текст может содержать строчные и заглавные буквы английского и русского алфавита, а также цифры.

`Формат выходных данных`

Программа должна вывести символ, который появляется наиболее часто.

Примечание 1. Если таких символов несколько, следует вывести последний по порядку символ.

Примечание 2. Следует различать заглавные и строчные буквы, а также буквы русского и английского алфавита.

In [31]:
n = input()
a, b = 0, 0
for i in n:
    if n.count(i) >= a:
        a = n.count(i)
        b = i
print(b)

abaabbcccc
c


# 50. Первое и последнее вхождение
На вход программе подается строка текста. Если в этой строке буква «f» встречается только один раз, выведите её индекс. Если она встречается два и более раз, выведите индекс её первого и последнего вхождения на одной строке, разделенных символом пробела. Если буква «f» в данной строке не встречается, следует вывести «NO».

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [32]:
n = input()
c = 0
a = 'f'
for i in n:
    if i in a:
        c += 1
if c == 1:
    print(n.find('f'))
elif c >= 2:    
    print(n.find('f'), n.rfind('f'))
else:
    print('NO')

abcdefg
5


# 51. Удаление фрагмента
На вход программе подается строка текста, в которой буква «h» встречается минимум два раза. Напишите программу, которая удаляет из этой строки первое и последнее вхождение буквы «h», а также все символы, находящиеся между ними.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [33]:
n = input()
print(n[0:n.find('h')], n[n.rfind('h') + 1:], sep='')

ahahahahaha
aa


# 52. Что покажет приведенный ниже фрагмент кода?
```
s = 'aabbAA111ccDDaa'
print(s.isalnum())
print(s.isalpha())
print(s.isdigit())
```


* True
* False
* False

```
True
True
False
```
```
False
False
False
```
```
True
True
True
```

# 53. Что покажет приведенный ниже фрагмент кода?
```
s = 'aabb!@#$11cc'
print(s.islower())
```


* False

* **True**

# 54. Что покажет приведенный ниже фрагмент кода?
```
s = 'AAb!@#$11CC'
print(s.isupper())
```


* **False**

* True

# 55. Что покажет приведенный ниже фрагмент кода?
```
s = '    abbc    '
print(s.isspace())
```


* **False**

* True

# 56. Дополните приведенный код, используя форматирование строк с помощью метода format, так чтобы он вывел текст: 

«In 2010, someone paid 10k Bitcoin for two pizzas.» (без кавычек).

In [34]:
s = 'In {0}, someone paid {1} {2} for two pizzas.'
a = 2010
b = '10k'
c = 'Bitcoin'
print('In {0}, someone paid {1} {2} for two pizzas.' .format(a,b,c))

In 2010, someone paid 10k Bitcoin for two pizzas.


# 57. Дополните приведенный код, используя форматирование строк с помощью f-строк, так чтобы он вывел текст: 

«In 2010, someone paid 10K Bitcoin for two pizzas.» (без кавычек).

In [35]:
year = 2010
amount = '10K'
currency = 'Bitcoin'

print(f'In {year}, someone paid {amount} {currency} for two pizzas.')

In 2010, someone paid 10K Bitcoin for two pizzas.


# 58. Что покажет приведенный ниже фрагмент кода?
```
print(ord('foo'))
```


* **Произойдет ошибка TypeError**

* 324

* 102 111 111

* 102

# 59. Символы в диапазоне
На вход программе подаются два числа a и b. Напишите программу, которая для каждого кодового значения в диапазоне от a до b (включительно), выводит соответствующий ему символ из таблицы символов Unicode.

`Формат входных данных` 

На вход программе подается два натуральных числа, каждое на отдельное строке.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [36]:
n, m = int(input()), int(input())
for i in range(n, m + 1):
    print(chr(i), end=' ')

65
70
A B C D E F 

# 60. Простой шифр
На вход программе подается строка текста. Напишите программу, которая переводит каждый ее символ в соответствующий ему код из таблицы символов Unicode.

`Формат входных данных` 

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести кодовые значения символов строки разделенных одним символом пробела.

In [37]:
n = input()
for i in range(len(n)):
    print(ord(n[i]), end=' ')

Hello world!
72 101 108 108 111 32 119 111 114 108 100 33 

# 61. Шифр Цезаря 🌶️
Легион Цезаря, созданный в 23 веке на основе Римской Империи не изменяет древним традициям и использует шифр Цезаря. Это их и подвело, ведь данный шифр очень простой. Однако в постапокалипсисе люди плохо знают все тонкости довоенного мира, поэтому ученые из НКР не могут понять как именно нужно декодировать данные сообщения. Напишите программу для декодирования этого шифра.

`Формат входных данных`

В первой строке дается число n (1≤ n≤ 25) – сдвиг, во второй строке даётся закодированное сообщение в виде строки со строчными латинскими буквами.

`Формат выходных данных`

Программа должна вывести одну строку – декодированное сообщение. Обратите внимание, что нужно декодировать сообщение, а не закодировать.

In [38]:
m, n = int(input()), input()
for i in n:
    d = ord(i) - m
    if d < 97:
        d += 26
    print(chr(d), end="") 

1
bwfusvfupdbftbs
avetruetocaesar